src/token_retriever/l1_retriever.rs

```diff
@@ -0,0 +1,142 @@
+use std::fmt;
+
+use http::{header::CONTENT_TYPE, StatusCode, Uri};
+use serde_json::json;
+
+use crate::{
+    authenticator::TokenRetrievalResponse,
+    http_client::{HttpClient, HttpClientError},
+    system_identity::input_data::auth_method::ClientSecret,
+    token::Token,
+    TokenRetriever, TokenRetrieverError,
+};
+
+/// HTTP-based token retriever for L1 authentication method (client ID + client secret)
+pub struct L1TokenRetriever<C: HttpClient> {
```

Comment:
Have you consider to Modify the current implementation of the TokenRetreiver and Authenticator to support L1 SI kinds? i think we might end reusing code if so. I think that the authenticator abstraction still make sense for L1 token retreival , so it would be consistent too.

Reply:
Discussed over call, changed in https://github.com/newrelic/newrelic-auth-rs/pull/34/commits/d7e76e8d3f6fc916af980b6be685638c1615434a!